client/core/core.go

```diff
@@ -1912,32 +1902,53 @@ func (c *Core) walletCheckAndNotify(w *xcWallet) bool {
 		return false
 	}
 	w.mtx.Lock()
+	wasSynced := w.synced
 	w.synced = synced
 	w.syncProgress = progress
 	w.mtx.Unlock()
 	c.notify(newWalletStateNote(w.state()))
-	if synced {
+	if synced && !wasSynced {
 		c.updateWalletBalance(w)
 		c.log.Infof("Wallet synced for asset %s", unbip(w.AssetID))
 	}
 	return synced
 }
 
-// monitorWalletSync repeatedly calls walletCheckAndNotify on a ticker until it
-// is synced. This should be run as a goroutine.
-func (c *Core) monitorWalletSync(ctx context.Context, wallet *xcWallet) {
-	ticker := time.NewTicker(syncTickerPeriod)
-	defer ticker.Stop()
-	for {
-		select {
-		case <-ticker.C:
-			if c.walletCheckAndNotify(wallet) {
+// startWalletSyncMonitor repeatedly calls walletCheckAndNotify on a ticker
+// until it is synced. This launches the monitor goroutine, if not already
+// running, and immediately returns.
+func (c *Core) startWalletSyncMonitor(wallet *xcWallet) {
+	// Prevent multiple sync monitors for this wallet.
+	if !atomic.CompareAndSwapUint32(&wallet.monitored, 0, 1) {
+		return // already monitoring
+	}
+
+	// If the wallet is shut down before sync is complete or Core stops, exit
+	// the loop by wrapping Core's context.
+	innerCtx, cancel := context.WithCancel(c.ctx)
+	go func() {
+		if wallet.connector.On() {
+			wallet.connector.Wait()
+		}
+		cancel()
+	}()
+	c.wg.Add(1)
+	go func() {
+		defer c.wg.Done()
+		defer atomic.StoreUint32(&wallet.monitored, 0)
+		ticker := time.NewTicker(syncTickerPeriod)
+		defer ticker.Stop()
+		for {
+			select {
+			case <-ticker.C:
+				if c.walletCheckAndNotify(wallet) {
+					return
```

Comment:
Not new behavior (see the old `connectWallet` method), but in this case the canceler goroutine above is stuck on `Wait` until the wallet shuts down even though the sync status loop is done.  Arguably a minor goroutine leak, but really just just excessive lifetime.  No issue IMO, but worth noting.